packages/client/docusaurus/docs/client/02-guides/01-authentication.mdx

```diff
@@ -3,8 +3,304 @@ id: 01-authentication
 title: Authenticating Your Users
 ---
 
-Stay tuned for more information on this topic.
+import SDKSpecific from '../SDKSpecific';
 
-[//]: # '- Authenticating Your Users'
-[//]: # '  - video/audio'
-[//]: # '  - with chat'
+## Introduction
+
+Stream uses JWT (JSON Web Tokens) for user authentication in our services.
+These tokens are valid across both Stream Video and Stream Chat services with the assumption
+that your App ID has both Chat and Video enabled on our [dashboard](https://dashboard.getstream.io/).
+
+For security reasons, we recommend that you generate the JWT token on your backend server as generating
+them on the client-side can lead to security issues.
+
+However, for development, you can use the following [JWT generator](https://getstream.io/chat/docs/react/token_generator/).
+
+:::warning
+Keep in mind to not expose your secret or your token to the public.
+:::
+
+## Token Providers
+
+At a high level, a Token Provider is a service, typically running on your backend, that generates a JWT for a given user.
+Usually, the Token Provider should perform the following tasks:
+
+- Check if the user is valid
+- Validate the user information against your system
+- Optionally, map your internal User ID to the Stream User ID
+- Generate, set expiry date, sign the JWT token and return it to the client
+
+You'll need to implement your own Token Provider prior to deploying your app to production.
+
+For more information, please refer to our [Server-Side Integration guide](../../basics/08-server-side-integrations).
+
+## Setting up a StreamVideoClient and connecting a user
+
+[`StreamVideoClient`](../04-call-engine/StreamVideoClient.md) is the main entry point for all the video-related features.
+In order to be able to construct it and connect a user, you'll need to be able to provide the following information:
+
+- `apiKey`: The API key for your Stream Video app
+- `tokenOrTokenProvider`: a static token or an `async` function that returns a token fetched from your Token Provider. In case when the token expires, our SDK will call this function again in order to get a fresh token.
+- `user`: An object that defines the user you want to connect. It should match the following type [`User`](../04-call-engine/User.md)
+
+```ts
```

Comment:
I'd put the hook code examples to the first place as most integrators would be using that.

Reply:
I'm also okay with not including LLC examples in React docs. I don't see a use case when someone should use them instead of the hooks